cloud-controller-manager/do/loadbalancers.go

```diff
@@ -184,8 +197,19 @@ func newLoadBalancers(resources *resources, client *godo.Client, region string)
 // GetLoadBalancer returns the *v1.LoadBalancerStatus of service.
 //
 // GetLoadBalancer will not modify service.
-func (l *loadBalancers) GetLoadBalancer(ctx context.Context, clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	lb, err := l.retrieveAndAnnotateLoadBalancer(ctx, service)
+func (l *loadBalancers) GetLoadBalancer(ctx context.Context, clusterName string, service *v1.Service) (lbs *v1.LoadBalancerStatus, truthy bool, err error) {
+	patcher := newServicePatcher(service)
+	defer func() {
+		if err == nil {
+			err = patcher.Patch(l.resources.kclient, service)
+			if err != nil {
+				err = fmt.Errorf("failed to patch service: %s", err)
```

Comment:
`patcher.Patch` returns annotated errors (that also include the Service name and namespace which this error currently doesn't), so we should return a bare `err` from that method call to avoid stuttering.

Reply:
If we go with patching regardless of the outer error (i.e., the value of `err` in line 203), we'll have to make sure that both an outer non-nil error and any non-nil error returned from `patcher.Patch` are propagated to the caller.

A convenient way to do that is to use [apimachinery's `Aggregate`](https://github.com/kubernetes/apimachinery/blob/release-1.15/pkg/util/errors/errors.go) helper: it basically aggregates (or wraps) multiple error values into a single error type that handles `nil` values appropriately and returns a single string representation of the errors. The usage in our case would look like this:

```go
import utilerrors "k8s.io/apimachinery/util/errors"
[...]

defer func() {
  perr := patcher.Patch(l.resources.kclient, service)
  errs := []errors{err, perr} // Errors can be nil; utilerrors.Aggregate will handle it correctly.
  err = utilerrors.NewAggregate(errs)
}
```

If you're curious, check out the [`Error()` implementation](https://github.com/kubernetes/apimachinery/blob/release-1.15/pkg/util/errors/errors.go#L64) to see what a string representation would look like.